.github/workflows/ci.yml

```diff
@@ -114,7 +117,7 @@ jobs:
           fi
 
           # Dependabot PRs are treated as if they are from forks (see above)
-          if [[ "$is_dependabot" == 'true' && "$is_default_branch" == 'false' ]]; then
+          if [[ "$is_dependabot" == 'true' && "${{ github.event_name }}" == 'pull_request' ]]; then
```

Comment:
This is a more specific check that we are running on a pull request, from the dependabot user